Sources/Starknet/Network/StarknetRequest.swift

```diff
@@ -1,25 +1,329 @@
 public struct StarknetRequest<U: Decodable> {
     let method: JsonRpcMethod
     let params: JsonRpcParams
-    let config: HttpNetworkProvider.Configuration
-    let networkProvider: HttpNetworkProvider
-
-    public func send() async throws -> U {
-        let rpcPayload = JsonRpcPayload(method: method, params: params)
-        let response = try await networkProvider.send(
-            payload: rpcPayload,
-            config: config,
-            receive: JsonRpcResponse<U>.self
-        )
-
-        if let error = response.error {
-            throw StarknetProviderError.jsonRpcError(error.code, error.message, error.data)
-        }
-
-        guard let result = response.result else {
-            throw StarknetProviderError.unknownError
-        }
-
-        return result
+}
+
+let defaultBlockId = StarknetBlockId.tag(.pending)
+let defaultSimulationFlagsForEstimateFee: Set<StarknetSimulationFlagForEstimateFee> = []
+
+public enum RequestBuilder {
+    /// Get the version of the Starknet JSON-RPC specification being used by the node.
+    ///
+    ///  - Returns: the version of the Starknet JSON-RPC specification being used.
+    static func getSpecVersion() -> StarknetRequest<String> {
```

Comment:
All the static methods need to be `public` otherwise they can't be used from outside this module.